lib/src/child-writer/index.js

```diff
@@ -56,20 +58,46 @@ const EXIT_CODES = require('../exit-codes');
 exports.write = (image, drive, options) => {
   const emitter = new EventEmitter();
 
-  utils.getTemporaryLogFilePath().then((logFile) => {
-    const argv = utils.getCLIWriterArguments({
-      entryPoint: rendererUtils.getApplicationEntryPoint(),
-      logFile: logFile,
-      image: image,
-      device: drive.device,
-      validateWriteOnSuccess: options.validateWriteOnSuccess,
-      unmountOnSuccess: options.unmountOnSuccess
-    });
+  const argv = utils.getCLIWriterArguments({
+    entryPoint: rendererUtils.getApplicationEntryPoint(),
+    image: image,
+    device: drive.device,
+    validateWriteOnSuccess: options.validateWriteOnSuccess,
+    unmountOnSuccess: options.unmountOnSuccess
+  });
+
+  // There might be multiple Etcher instances running at
+  // the same time, therefore we must ensure each IPC
+  // server/client has a randomly generated name.
+  const RANDOM_STRING = crypto.randomBytes(20).toString('hex');
+  process.env.IPC_SERVER_ID = `etcher-server-${RANDOM_STRING}`;
+  process.env.IPC_CLIENT_ID = `etcher-client-${RANDOM_STRING}`;
+
+  ipc.config.id = process.env.IPC_SERVER_ID;
+  ipc.config.silent = true;
+  ipc.serve();
 
-    // Make writer proxy inherit the temporary log file location
-    // while keeping current environment variables intact.
-    process.env[CONSTANTS.TEMPORARY_LOG_FILE_ENVIRONMENT_VARIABLE] = logFile;
+  ipc.server.on('error', (error) => {
+    emitter.emit('error', error);
+  });
+
+  ipc.server.on('message', (data) => {
+    const message = JSON.parse(data);
 
+    // The error object is decomposed by the CLI for serialisation
+    // purposes. We compose it back to an `Error` here in order
+    // to provide better encapsulation.
+    if (message.command === 'error') {
+      const error = new Error(message.data.message);
```

Comment:
Doesn't this mean that the call stack will be lost (the error object will populate the stack dump with the location of this code)? The error reporting here should be focused on pretty intently, because this is by far the most common issue I see people have with Etcher: Cryptic errors bubbling up from the CLI.